dbms/src/Flash/Coprocessor/RemoteRequest.cpp

```diff
@@ -125,13 +125,37 @@ RemoteRequest RemoteRequest::build(
         tici_scan.constructTiCIScanForRemoteRead(mutable_tici_scan);
 
         const auto & return_columns = tici_scan.getReturnColumns();
-        auto names_and_types = genNamesAndTypesForTiCI(return_columns, "column");
-        for (size_t i = 0; i < return_columns.size(); ++i)
+        NamesAndTypes names_and_types;
+        if (tici_scan.isCount())
         {
-            const auto & col = return_columns[i];
-            schema.emplace_back(std::make_pair(names_and_types[i].name, col));
-            dag_req.add_output_offsets(i);
+            names_and_types = tici_scan.getNamesAndTypes();
+            TiDB::ColumnInfo ci;
+            // count query always returns a NOT NULL INT64 column
+            ci.tp = TiDB::TypeLongLong;
+            ci.setNotNullFlag();
+            schema.emplace_back(std::make_pair(names_and_types[0].name, std::move(ci)));
+            dag_req.add_output_offsets(0);
+            // set index_id to -1 to indicate that it is a count query
+            mutable_tici_scan->mutable_fts_query_info()->set_index_id(-1);
```

Comment:
I think using index_id = -1 to determine whether it is a count is not ideal. Should we consider adding an is_count field in ftsinfo instead?